service/src/main/java/bio/terra/cbas/dao/RunSetDao.java

```diff
@@ -81,25 +82,45 @@ public int updateLastPolled(List<UUID> runSetIds) {
   }
 
   public int updateStateAndRunDetails(
-      UUID runSetId, CbasRunSetStatus newStatus, Integer runCount, Integer errorCount) {
+      UUID runSetId,
+      CbasRunSetStatus newStatus,
+      Integer runCount,
+      Integer errorCount,
+      OffsetDateTime lastModified) {
     OffsetDateTime currentTimestamp = DateUtils.currentTimeInUTC();
-    String sql =
-        "UPDATE run_set SET status = :status, last_modified_timestamp = :last_modified_timestamp, last_polled_timestamp = :last_polled_timestamp, run_count = :run_count, error_count = :error_count WHERE run_set_id = :run_set_id";
+
+    String updateClause =
+        "UPDATE run_set SET %s = :status, %s = :last_polled_timestamp, %s = :run_count, %s = :error_count"
+            .formatted(
+                RunSet.STATUS_COL,
+                RunSet.LAST_POLLED_TIMESTAMP_COL,
+                RunSet.RUN_COUNT_COL,
+                RunSet.ERROR_COUNT_COL);
+
+    if (lastModified != null) {
+      updateClause =
+          updateClause + ", %s = :last_modified".formatted(RunSet.LAST_MODIFIED_TIMESTAMP_COL);
+    }
+
+    String sql = updateClause + " WHERE %s = :run_set_id".formatted(RunSet.RUN_SET_ID_COL);
     return jdbcTemplate.update(
         sql,
         new MapSqlParameterSource(
             Map.of(
-                RunSet.RUN_SET_ID_COL,
+                "run_set_id",
                 runSetId,
-                RunSet.STATUS_COL,
+                "status",
                 newStatus.toString(),
-                RunSet.LAST_MODIFIED_TIMESTAMP_COL,
+                "last_polled_timestamp",
                 currentTimestamp,
-                RunSet.LAST_POLLED_TIMESTAMP_COL,
-                currentTimestamp,
-                RunSet.RUN_COUNT_COL,
+                "run_count",
                 runCount,
-                RunSet.ERROR_COUNT_COL,
-                errorCount)));
+                "error_count",
+                errorCount,
+                "last_modified",
+                // If we hit the orelse, we shouldn't be adding lastModified to the DB query, so MAX
+                // should never actually be hitting
+                // the database. If it does, we have a bug!)
```

Comment:
just to make sure I understand - you're saying that the `.orElse` should never actually be executed, if things are working properly?

does that mean it would be better to check whether `lastModified` is null first, and throw a descriptive error?

Reply:
The null is handled earlier in the function by not including it in the template. I suppose I could also not include it in the map unless it's not null either, to avoid the risk of database corruption if the logic is wrong. Let me try that